File: examples/guessnum/interactor.cpp
#include <bits/stdc++.h>
#include "testlib.h"
using namespace std;

const int MAX_ABS = 1'000'000'000;

int main(int argc, char **argv){
    registerInteraction(argc, argv);

    int target = inf.readInt();
    int moves = inf.readInt();
    if (moves < 0) quitf(_fail, "invalid number of moves!");
    cout << moves << endl;
    while (true) {
        string tok = ouf.readToken();
        int value = ouf.readInt(-MAX_ABS, MAX_ABS);
        tout << tok << " " << value << endl;
        if (tok == "answer") {
            quitf(_ok, "got the answer. pass to checker");
        } else if (tok == "ask") {
            if (moves == 0) quitf(_wa, "ran out of moves");
            moves--;
            cout << value - target << endl;
        } else {
            quitf(_wa, "unknown action: %s", tok.c_str());
        }
    }
}

File: examples/guessnum/interactor.py
from sys import *
from kg.interactors import * ### @import

MAX_ABS = 10**9

lim = Bounds(value = abs(+Var) <= MAX_ABS)

@interactor
def interact(input_stream, user_stream, *, output_stream=None, **kwargs):
    [target, moves] = input_stream.read.int().space.int().eoln
    if moves < 0: raise Fail("invalid number of moves!")
    user_stream.print(moves)
    while True:
        [tok, value] = user_stream.read.token().space.int(lim.value).eoln
        if output_stream: output_stream.print(tok, value)
        if tok == "answer":
            return 1.0 # got the answer. pass to checker
        elif tok == "ask":
            if moves == 0: raise Wrong("ran out of moves")
            moves -= 1
            user_stream.print(value - target)
        else:
            raise Wrong(f"unknown action: {tok}")


if __name__ == '__main__':
    interact_with(interact)


